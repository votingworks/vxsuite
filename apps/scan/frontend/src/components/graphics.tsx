import React from 'react';
import styled from 'styled-components';
import { Graphic } from '@votingworks/ui';

export const PlaceholderGraphic = styled.div`
  margin: 0 auto 1rem;
  border-radius: 1rem;
  background: #6638b6;
  width: 250px;
  height: 250px;
`;

export function QuestionCircle(): JSX.Element {
  return (
    <Graphic
      src="/assets/question-circle.svg"
      alt="question-circle"
      aria-hidden
    />
  );
}

export function DoNotEnter(): JSX.Element {
  return (
    <Graphic src="/assets/do-not-enter.svg" alt="do-not-enter" aria-hidden />
  );
}

export function CircleCheck(): JSX.Element {
  return (
    <Graphic src="/assets/check-circle.svg" alt="check-circle" aria-hidden />
  );
}

export function TimesCircle(): JSX.Element {
  return (
    <Graphic
      src="/assets/times-circle.svg"
      alt="times-circle"
      aria-hidden
      style={{ height: '400px' }}
    />
  );
}

export function ExclamationTriangle(): JSX.Element {
  return (
    <Graphic
      src="/assets/exclamation-triangle.svg"
      alt="exclamation-triangle"
      aria-hidden
    />
  );
}
