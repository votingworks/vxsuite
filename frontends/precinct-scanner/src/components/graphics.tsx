import React from 'react';
import styled from 'styled-components';

export const PlaceholderGraphic = styled.div`
  margin: 0 auto 1rem;
  border-radius: 1rem;
  background: #6638b6;
  width: 250px;
  height: 250px;
`;

export const Graphic = styled.img`
  margin: 0 auto 1rem;
  width: 250px;
`;

export function QuestionCircle(): JSX.Element {
  return (
    <Graphic
      src={`${process.env.PUBLIC_URL}/assets/question-circle.svg`}
      alt="question-circle"
      aria-hidden
    />
  );
}

export function DoNotEnter(): JSX.Element {
  return (
    <Graphic
      src={`${process.env.PUBLIC_URL}/assets/do-not-enter.svg`}
      alt="do-not-enter"
      aria-hidden
    />
  );
}

export function CircleCheck(): JSX.Element {
  return (
    <Graphic
      src={`${process.env.PUBLIC_URL}/assets/check-circle.svg`}
      alt="check-circle"
      aria-hidden
    />
  );
}

export function TimesCircle(): JSX.Element {
  return (
    <Graphic
      src={`${process.env.PUBLIC_URL}/assets/times-circle.svg`}
      alt="times-circle"
      aria-hidden
      style={{ height: '400px' }}
    />
  );
}

export function ExclamationTriangle(): JSX.Element {
  return (
    <Graphic
      src={`${process.env.PUBLIC_URL}/assets/exclamation-triangle.svg`}
      alt="exclamation-triangle"
      aria-hidden
    />
  );
}

export function InsertBallot(): JSX.Element {
  return (
    <Graphic
      src={`${process.env.PUBLIC_URL}/assets/insert-ballot.svg`}
      alt="insert-ballot"
      aria-hidden
      style={{ width: '600px' }}
    />
  );
}

export function IndeterminateProgressBar(): JSX.Element {
  return (
    <Graphic
      src={`${process.env.PUBLIC_URL}/assets/indeterminate-progress-bar.svg`}
      alt="indeterminate-progress-bar"
      aria-hidden
      style={{ width: '400px' }}
    />
  );
}

export function RotateCard(): JSX.Element {
  return (
    <Graphic
      src={`${process.env.PUBLIC_URL}/assets/rotate-card.svg`}
      alt="rotate-card"
      aria-hidden
      style={{ width: '300px' }}
    />
  );
}
