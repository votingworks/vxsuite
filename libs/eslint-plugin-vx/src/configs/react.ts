import recommended = require('./recommended');

const jsExtensions = ['.js', '.jsx'];
const tsExtensions = ['.ts', '.tsx'];
const allExtensions = jsExtensions.concat(tsExtensions);

export = {
  ...recommended,
  extends: [
    ...recommended.extends
      .map((name) => (name === 'airbnb-base' ? 'airbnb' : name))
      .filter((name) => name !== 'plugin:prettier/recommended'),
    'airbnb/hooks',
    'plugin:react/recommended',
    'plugin:jsx-a11y/recommended',
    'plugin:prettier/recommended', // Enables eslint-plugin-prettier and displays prettier errors as ESLint errors. Make sure this is always the last configuration in the extends array.
  ],
  settings: {
    ...recommended.settings,
    react: {
      version: 'detect',
    },
  },
  rules: {
    ...recommended.rules,
    'jsx-a11y/label-has-associated-control': [
      'error',
      {
        controlComponents: ['Select'],
      },
    ],
    'react/jsx-filename-extension': ['error', { extensions: allExtensions }],
    'react/jsx-fragments': ['error', 'element'],
    'react/jsx-no-bind': 'off',
    'react/jsx-no-constructed-context-values': 'off',
    'react/jsx-no-useless-fragment': ['error', { allowExpressions: true }],
    'react/jsx-one-expression-per-line': 'off',
    'react/jsx-props-no-spreading': 'off',
    'react/jsx-wrap-multilines': 'off',
    'react/prop-types': 'off',
    'react/require-default-props': 'off',
  },
};
