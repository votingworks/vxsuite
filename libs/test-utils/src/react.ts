/* eslint-disable no-console -- this file monkey-patches `console` */

/**
 * Suppresses the "Can't perform a React state update on an unmounted component" warning by
 * modifying the `console.error` method to ignore the warning.
 *
 * This is a temporary workaround until we upgrade to React 18.
 */
export function suppressReact17UnmountedWarning(): void {
  const originalConsoleError = console.error;
  console.error = (...args: unknown[]) => {
    if (
      typeof args[0] === 'string' &&
      args[0].startsWith(
        `Warning: Can't perform a React state update on an unmounted component`
      )
    ) {
      return;
    }

    originalConsoleError(...args);
  };
}
