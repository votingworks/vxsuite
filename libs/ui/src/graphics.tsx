import React from 'react';
import styled from 'styled-components';

export const Graphic = styled.img`
  margin: 0 auto 1rem;
  width: 250px;
`;

const UsbImageStyles = styled.img`
  margin-right: auto;
  margin-left: auto;
  height: 200px;
`;

export function UsbImage(): JSX.Element {
  return <UsbImageStyles src="/assets/usb-drive.svg" alt="Insert USB Image" />;
}

export function IndeterminateProgressBar(): JSX.Element {
  return (
    <Graphic
      src="/assets/indeterminate-progress-bar.svg"
      alt="indeterminate-progress-bar"
      aria-hidden
      style={{ width: '400px' }}
    />
  );
}
