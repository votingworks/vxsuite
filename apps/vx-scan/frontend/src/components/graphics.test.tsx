// This is a temporary test to be removed once SVG graphics are created.
import React from 'react';
import { render } from '@testing-library/react';
import { PlaceholderGraphic } from './graphics';

test('Renders Absolute top right', () => {
  const { container } = render(<PlaceholderGraphic />);
  expect(container).toMatchSnapshot();
});
