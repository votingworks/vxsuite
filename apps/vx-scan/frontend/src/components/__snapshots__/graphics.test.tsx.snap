// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Absolute top right 1`] = `
<div>
  <div
    class="sc-bdvvaa fPyvOy"
  />
</div>
`;
